mypy/newsemanal/semanal.py

```diff
@@ -4529,7 +4530,9 @@ def is_local_name(self, name: str) -> bool:
     def fail(self,
              msg: str,
              ctx: Context,
-             serious: bool = False, *,
+             serious: bool = False,
+             *,
+             code: Optional[ErrorCode] = None,
```

Comment:
We should add the code argument to the versions of this method in SemanticAnalyzerCoreInterface and maybe in TypeAnalyzerPluginInterface.

Reply:
Yes. I'd rather do this only after we've removed the old semantic analyzer, since changing this would require changes in it.